src/lib/math/bigint/big_code.cpp

```diff
@@ -140,19 +110,26 @@ secure_vector<uint8_t> BigInt::encode_fixed_length_int_pair(const BigInt& n1, co
       throw Encoding_Error("encode_fixed_length_int_pair: values too large to encode properly");
    }
    secure_vector<uint8_t> output(2 * bytes);
-   n1.binary_encode(output.data(), bytes);
-   n2.binary_encode(output.data() + bytes, bytes);
+   n1.serialize_to(std::span{output}.subspan(0, bytes));
+   n2.serialize_to(std::span{output}.subspan(bytes, bytes));
    return output;
 }
 
+BigInt BigInt::decode(std::span<const uint8_t> buf, Base base) {
+   if(base == Binary) {
+      return BigInt::from_bytes(buf);
+   }
+   return BigInt::decode(buf.data(), buf.size(), base);
```

Comment:
For these functions using `BigInt::Base` I'm inclined to not do anything with `span` since already the whole set of functions and the enum itself are deprecated. We'd just be introducing new interfaces which begin life deprecated, which seems odd.